File: applets/src/org/rr0/ufoathome/model/ufo/UFOSceneModel.java
package org.rr0.ufoathome.model.ufo;

import org.rr0.ufoathome.view.gui.MapElement;
import org.rr0.ufoathome.model.sky.SkyModel;

import java.util.Hashtable;

/**
 * @author Jerôme Beau
 * @version 15 nov. 2003 19:38:01
 */
public class UFOSceneModel extends SkyModel {
    private Hashtable buildings = new Hashtable();
    protected Hashtable descriptions = new Hashtable();
    protected WitnessModel witness = new WitnessModel();

    public UFO createUFO(String name) {
        UFO ufo = new UFO(name);
        sources.put(name, ufo);
        return ufo;
    }

    public MapElement getBuidling(String objectName) {
        return (MapElement) buildings.get(objectName);
    }

    public void setDescription(String timeKey, String description) {
        descriptions.put(timeKey, description);
    }

    public String getDescription(String timeKey) {
        return (String) descriptions.get(timeKey);
    }

    public WitnessModel getWitness() {
        return witness;
    }
}

File: applets/src/org/rr0/ufoathome/model/ufo/UFOSceneModel.ts
import {UFO} from "./UFO";
import {SkyModel} from "../sky/SkyModel";
import {WitnessModel} from "./WitnessModel";

export class UFOSceneModel extends SkyModel {
  private buildings = new Hashtable();
  protected descriptions = new Hashtable();
  protected witness = new WitnessModel();

  public createUFO(name: String): UFO {
    const ufo = new UFO(name);
    this.sources.put(name, ufo);
    return ufo;
  }

  public getBuidling(objectName: String): MapElement {
    return <MapElement>this.buildings.get(objectName);
  }

  public setDescription(timeKey: String, description: String) {
    this.descriptions.put(timeKey, description);
  }

  public getDescription(timeKey: String): String {
    return <String> this.descriptions.get(timeKey);
  }

  public getWitness(): WitnessModel {
    return this.witness;
  }
}
